src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java

```diff
@@ -174,8 +173,8 @@ public void reportCppComment(int startLineNo, int startColNo) {
      * the value is the comment {@link TextBlock} at the line.
      * @return the Map of comments
      */
-    public ImmutableMap<Integer, TextBlock> getSingleLineComments() {
-        return ImmutableMap.copyOf(cppComments);
+    public Map<Integer, TextBlock> getSingleLineComments() {
+        return new HashMap<>(cppComments);
```

Comment:
Should it be
`        return Collections.unmodifiableMap(cppComments);
`
?

Reply:
Fixed. thanks @pbludov  .